docs/email-setup.md

```diff
@@ -0,0 +1,266 @@
+  groupId: 'your-group',
+  notificationType: 'WELCOME'
+})
+```
+
+**Example from Foam**: See `foam3/deployment/test/notificationTemplates.jrl` for notification template examples.
+
+## Step 3: Configure Notification Settings
+
+1. Create a new file `deployment/your-environment/notificationSettings.jrl`:
+
+```javascript
+p({
+  class: 'foam.core.notification.EmailSetting',
+  id: 'user-email-setting',
+  owner: userId,
+  enabled: true
+})
+```
+
+**Example from Foam**: See `foam3/deployment/test/notificationSettings.jrl` for settings examples.
+
+## Step 4: Implement Email Notifications
+
+### Option 1: Using Notification DAO Directly
+
+Create a new Java class in your project:
+
+```java
+package your.package.notification;
+
+import foam.core.X;
+import foam.dao.DAO;
+import foam.core.notification.Notification;
+import java.util.HashMap;
+import java.util.Map;
+
+public class WelcomeEmailService {
+    private final DAO notificationDAO;
+
+    public WelcomeEmailService(X x) {
+        this.notificationDAO = (DAO) x.get("notificationDAO");
+    }
+
+    public void sendWelcomeEmail(X x, long userId) {
+        Notification notification = new Notification();
+        notification.setTemplate("welcome-notification");
+        notification.setEmailName("welcome-email");
+        notification.setBody("Welcome to our application!");
+        notification.setUserId(userId);
+
+        Map<String, Object> emailArgs = new HashMap<>();
+        emailArgs.put("templateSource", this.getClass().getName());
+        notification.setEmailArgs(emailArgs);
+
+        notificationDAO.put_(x, notification);
+    }
+}
+```
+
+**Example from Foam**: See `foam3/src/foam/core/notification/test/EmailNotificationTest.java` for a complete example.
+
+### Option 2: Using DAO Notification Rules
+
+1. Create a new rule action class:
+
+```java
+package your.package.notification;
+
+import foam.core.notification.DAONotificationRuleAction;
+import foam.core.X;
+import foam.core.FObject;
+import foam.core.Rule;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserCreatedNotificationRuleAction extends DAONotificationRuleAction {
+    @Override
+    public void applyAction(X x, FObject obj, FObject oldObj, Rule rule) {
+        if (!(obj instanceof User)) return;
+
+        User user = (User) obj;
+        Map<String, Object> args = new HashMap<>();
+        args.put("userName", user.getLegalName());
+        args.put("templateSource", this.getClass().getName());
+
+        Notification notification = new Notification();
+        notification.setTemplate("welcome-notification");
+        notification.setEmailName("welcome-email");
+        notification.setBody("Welcome " + user.getLegalName() + "!");
+        notification.setUserId(user.getId());
+        notification.setEmailArgs(args);
+
+        ((DAO) x.get("notificationDAO")).put_(x, notification);
+    }
+}
+```
+
+2. Register the rule in `deployment/your-environment/daoConfig.jrl`:
```

Comment:
@abdelaziz-mahdy I’m not familiar with daoConfig. 
Normally rules are added via rules.jrl.  This is an example matching your documentation:

2. Register the rule in `deployment/your-evironment/rules.jrl`:
```javascript
p({
  "class":"foam.core.ruler.Rule",
  "id":"your-package-notification-UserCreatedNotificationRule",
  "name":"UserCreatedNotificationRule",
  "documentation":"Generate notification to new user on user creation",
  "enabled":true,
  "ruleGroup": "auth",
  "daoKey": "localUserDAO",
  "operation": 0,
  "lifecycleState": 1,
  action: {
    "class": "your.package.notification.UserCreatedNotificationRuleAction"
  }
})
```

Reply:
pr: https://github.com/kgrgreer/foam3/pull/3985 with the changes, thank you